### DISCUSSION TITLE
Quarkus 2 and Agroal : WARN Closing connection in incorrect state CHECKED_OUT

### DISCUSSION THREAD

**Discussion by yblazart** [quarkusio/quarkus#24281] [Community]
Hello, I'm currently working on a very big project (in a big company of course). 
We were forced to stay on Quarkus 1.13.7 because of jdk 8. 
Now we can pass to jdk 11, so I attempt to migrate my project direcly to Quarkus 2.7.4.

I precise that I'm writing this on my personnal account, as you can certainly guess I don't have rights to do it from the company network and I can't also push code.

I'm facing a lot of WARN messages from agroal "Closing connection in incorrect state CHECKED_OUT'" when I start mvn quarkus:dev.

In fact I have some (a lot) of Schedules that play with JPA. Using normal @Transactionnal. and so on.
I just isolated one to check and the effect is the same.

I attempted to understand while debuging, but it seems that intellij don't show me the right code from agroal, breakpoints are not in front of good lines.

Another information is that my database kind is 'other', because I need to use h2 for test and for Non regression tests on jenkins (so my microsservices are launched as in "real" but with H2 url), but for delivery we use Oracle of course. I'm aware that it's not certainly a good way of doing that, but as you can guess, we did this with time we got, and constraints.

I will try to reproduce that behaviour on my own device, but it can take time, and I'm not sure to success. 

So if anybody have an idea ? Or see immediately what's happening, help is welcomed.

Long life to Quarkus, and thanks !

**Comment by yblazart** [comment]
I just try again specifiyng db kind to h2 and put quarkus-jdbc-h2, same effect....

**Reply by maxandersen** [reply]
What agroal / database configuration are you running with ? Which properties ?

**Reply by yblazart** [reply]
Sorry for the delay. 

quarkus.datasource.jdbc.urL=jdbc:h2:tcp://Zocalhost:59090/xxx;MODE=OracLe;SCHEMA=XXX;
quarkus.datasource.username=sa
quarkus.datasource.password=sa
guarkus.datasource.jdbc.driver=org.h2.Driver
quarkus.hibernate-orm.dialect=org.hibernate.dialect.H2Dialect
quarkus.datasource.jdbc.transaction-isolation-level=read-committed
quarkus.datasource.jdbc.flush-on-close=true

%dev.quarkus.datasource.jdbc.url=jdbc:h2:tcp://Localhost:59090/xxxx;MODE=Oracle;SCHEMA=xxxx
%dev.quarkus.datasource.username=sa
%dev.quarkus.datasource.password=sa
%dev.quarkus.datasource.jdbc.driver=org.h2.Driver
%dev.quarkus.hibernate-orm.dialect=org.hibernate.dialect.H2DiaLect

**Comment by yblazart** [comment]
Ok, while removing these properties, it's good now. But why ? Sorry... And thanks for help.

(I think it's more related to flus on close)

quarkus.datasource.jdbc.transaction-isolation-level=read-committed
quarkus.datasource.jdbc.flush-on-close=true

**Comment by gsmet** [comment]
@barreiro could you have a look at this discussion?

**Reply by barreiro** [reply]
Thanks for bringing this to my attention. Will try to reproduce.

**Comment by barreiro** [comment]
Thank you very much for reporting @yblazart. This was indeed an issue in Agroal with the flush-on-close state transition. 

Fixed in [AG-189](https://issues.redhat.com/browse/AG-189)

**Reply by PStoner3** [reply]
How long, do you you suppose, until this is released?

**Reply by gsmet** [reply]
@barreiro any chance you could release a new Agroal version before Monday evening?
